plugins/modules/nd_api_key.py

```diff
@@ -0,0 +1,180 @@
+    state: query
+  register: query_result_by_id
+
+- name: Query all API keys
+  cisco.nd.nd_api_key:
+    state: query
+  register: all_keys
+
+- name: Delete an API key by ID
+  cisco.nd.nd_api_key:
+    api_key_id: "{{ result.current.id }}"
+    state: absent
+
+- name: Delete an API key by name
+  cisco.nd.nd_api_key:
+    api_key_name: "ansible_test_key"
+    state: absent
+"""
+
+RETURN = r"""
+"""
+
+from copy import deepcopy
+import re
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.nd.plugins.module_utils.nd import NDModule, nd_argument_spec
+
+
+def main():
+    argument_spec = nd_argument_spec()
+    argument_spec.update(
+        api_key_id=dict(type="str"),
+        api_key_name=dict(type="str", aliases=["description"]),
+        annotations=dict(type="dict"),
+        state=dict(type="str", default="present", choices=["present", "absent", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "present", ["api_key_name"]],
+            ["state", "absent", ["api_key_name", "api_key_id"], True],
+        ],
+    )
+
+    nd = NDModule(module)
+
+    api_key_id = nd.params.get("api_key_id")
+    api_key_name = nd.params.get("api_key_name")
+    annotations = nd.params.get("annotations")
+    state = nd.params.get("state")
+
+    if annotations is None:
+        annotations = {}
+
+    path = "/api/v1/infra/aaa/apiKeys"
+    if api_key_id:
+        nd.existing = nd.previous = deepcopy(nd.query_obj("{0}/{1}".format(path, api_key_id)))
+    elif api_key_name:
+        nd.existing = nd.previous = deepcopy(nd.get_obj(path, key="apiKeys", apiKeyName=api_key_name))
+    else:
+        nd.existing = nd.previous = nd.query_objs(path, key="apiKeys")
+
+    if state == "present":
+
+        if len(api_key_name) > 32 or len(api_key_name) < 1:
+            nd.fail_json("A length of 1 to 32 characters is allowed.")
+        elif re.search(r"[^a-zA-Z0-9_.-]", api_key_name):
+            nd.fail_json("API Key name contains invalid characters. Valid characters include letters, digits, '_', '.', and '-'.")
+
+        payload = {
+            "apiKeyName": api_key_name,
+            "annotations": annotations,
+        }
+
+        nd.sanitize(payload)
+
+        if not module.check_mode and nd.existing:
+            if nd.existing.get("apiKeyName") != api_key_name:
+                update_path = "{0}/{1}".format(path, api_key_id)
+                nd.request(update_path, method="PUT", data=payload)
+                nd.existing = nd.query_obj(update_path)
+        elif not nd.existing:
+            nd.existing = nd.request(path, method="POST", data=payload)
+        else:
+            nd.existing = nd.proposed
+
+    elif state == "absent":
+        if nd.existing and not module.check_mode:
+            nd.request("{0}/{1}".format(path, nd.existing["id"]), method="DELETE")
+            nd.existing = {}
```

Comment:
should the existing be set to {} when check_mode is true?

```suggestion
        if nd.existing and not module.check_mode:
            nd.request("{0}/{1}".format(path, nd.existing["id"]), method="DELETE")
        nd.existing = {}
```